File: src/main/scala/com/vc/easy/L896.scala
package com.vc.easy

object L896 {
  def isMonotonic(arr: Array[Int]): Boolean = {
    val n = arr.length
    var asc = true
    var desc = true
    var i = 1
    while(i < n) {
      if(arr(i) > arr(i - 1)) {
        desc = false
        if(!asc) return false
      }
      if(arr(i) < arr(i - 1)) {
        asc = false
        if(!desc) return false
      }
      i += 1
    }
    true
  }
}

File: src/main/java/com/vc/easy/L896.java
package com.vc.easy;

class L896 {
    public boolean isMonotonic(int[] A) {
        if(A == null || A.length <= 1) return true;

        boolean increasing = false, decreasing = false;
        for(int i = 0; i < A.length - 1; i++) {
            int current = A[i];
            int next = A[i + 1];
            int compare = Integer.compare(current, next);
            if(compare > 0) decreasing = true;
            else if(compare < 0) increasing = true;
        }
        return !(increasing && decreasing);
    }
}
